source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -244,14 +269,103 @@ bool Filter::parseExtensions(os_fd_t fd) {
     if (bytes_avail == 0) {
       return false;
     }
-    bytes_avail = std::min(size_t(bytes_avail), sizeof(buf_));
+    bytes_avail = std::min(size_t(bytes_avail), buf_size);
     bytes_avail = std::min(size_t(bytes_avail), proxy_protocol_header_.value().extensions_length_);
-    const Api::SysCallSizeResult recv_result = os_syscalls.recv(fd, buf_, bytes_avail, 0);
+    buf += buf_off ? *buf_off : 0;
+    const Api::SysCallSizeResult recv_result = os_syscalls.recv(fd, buf, bytes_avail, 0);
     if (recv_result.rc_ != bytes_avail) {
       throw EnvoyException("failed to read proxy protocol extension");
     }
     proxy_protocol_header_.value().extensions_length_ -= recv_result.rc_;
+
+    if (buf_off) {
+      *buf_off += recv_result.rc_;
+    }
   }
+
+  return true;
+}
+
+/**
+ * @note  A TLV is arranged in the following format: @n
+ *        1st byte is type, @n
```

Comment:
I think it would be clearer to copy/paste the struct definition from http://www.haproxy.org/download/1.8/doc/proxy-protocol.txt (and attribute to that page)

Reply:
updated as suggested.